src/Features/Core/Completion/AbstractCompletionService.cs

```diff
@@ -130,65 +135,146 @@ private class ProviderGroup
             }
 
             // Now, ask all the triggered providers if they can provide a group.
-            var providersAndGroups = new List<ProviderGroup>();
-            foreach (var p in triggeredProviders)
+            var providersAndLists = new List<ProviderList>();
+            foreach (var provider in triggeredProviders)
             {
-                var g = await GetGroupAsync(p, documentOpt, text, position, triggerInfo, cancellationToken).ConfigureAwait(false);
-                if (g != null)
+                var completionList = await GetCompletionListAsync(provider, documentOpt, text, position, triggerInfo, cancellationToken).ConfigureAwait(false);
+                if (completionList != null)
                 {
-                    providersAndGroups.Add(new ProviderGroup { Provider = p, Group = g });
+                    providersAndLists.Add(new ProviderList { Provider = provider, List = completionList });
                 }
             }
 
             // See if there was a group provided that was exclusive and had items in it.  If so, then
             // that's all we'll return.
-            var firstExclusiveGroup = providersAndGroups.FirstOrDefault(
-                t => t.Group.IsExclusive && t.Group.Items.Any());
+            var firstExclusiveList = providersAndLists.FirstOrDefault(
+                t => t.List.IsExclusive && t.List.Items.Any());
 
-            if (firstExclusiveGroup != null)
+            if (firstExclusiveList != null)
             {
-                return SpecializedCollections.SingletonEnumerable(firstExclusiveGroup.Group);
+                return MergeAndPruneCompletionLists(SpecializedCollections.SingletonEnumerable(firstExclusiveList.List), completionRules);
             }
 
             // If no exclusive providers provided anything, then go through the remaining
             // triggered list and see if any provide items.
-            var nonExclusiveGroups = providersAndGroups.Where(t => !t.Group.IsExclusive).ToList();
+            var nonExclusiveLists = providersAndLists.Where(t => !t.List.IsExclusive).ToList();
 
             // If we still don't have any items, then we're definitely done.
-            if (!nonExclusiveGroups.Any(g => g.Group.Items.Any()))
+            if (!nonExclusiveLists.Any(g => g.List.Items.Any()))
             {
                 return null;
             }
 
             // If we do have items, then ask all the other (non exclusive providers) if they
             // want to augment the items.
```

Comment:
Well, augmentation still happens, but the property isn't used. Essentially, if a completion list is going to be displayed, all the providers get called again to contribute. That's the way it's always worked.